watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskFragment.java

```diff
@@ -113,23 +104,43 @@ public void onRefresh() {
                 getTasksRequest();
             }
         });
-    }
 
-    /**
-     * Starts a new fragment for the corresponding task view
-     * @param l - ListView that was clicked
-     * @param v - View that was clicked
-     * @param position - Position of view
-     * @param id = id of View
-     */
-    @Override
-    public void onListItemClick(ListView l, View v, int position, long id) {
-        Task taskClicked;
-        if (TASK_TYPE == USER) taskClicked = mUserTaskList.get(position);
-        else taskClicked = mAllTaskList.get(position);
+        mListView = (ExpandableListView) view.findViewById(android.R.id.list);
+        mListView.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
+            @Override
+            public boolean onChildClick(ExpandableListView expandableListView, View view, int groupPosition, int childPosition, long l) {
+                Log.i("asdf", "clicked");
+                Task taskClicked;
+                if (mArgs.getInt(OPTION) == USER) taskClicked = mUserTaskList.get(mTaskListHeaders.get(groupPosition)).get(childPosition);
```

Comment:
Is this the best way to check whether the click in on the user task list or the all task list?